pkg/transport/generic_transport_controller.go

```diff
@@ -0,0 +1,512 @@
+	wrappedObjectGVR, err := getGvrFromWrappedObject(transportRestConfig, emptyWrappedObject)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get transport wrapped object GVR - %w", err)
+	}
+
+	dynamicInformer := dynamicinformer.NewDynamicSharedInformerFactory(transportDynamicClient, 0)
+	wrappedObjectGenericInformer := dynamicInformer.ForResource(*wrappedObjectGVR)
+
+	transportController := &genericTransportController{
+		logger:                          klog.FromContext(ctx).WithName(ControllerName),
+		placementDecisionLister:         placementDecisionInformer.Lister(),
+		placementDecisionInformerSynced: placementDecisionInformer.Informer().HasSynced,
+		wrappedObjectInformerSynced:     wrappedObjectGenericInformer.Informer().HasSynced,
+		workqueue:                       workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName),
+		transport:                       transport,
+		transportClient:                 transportDynamicClient,
+		wrappedObjectGVR:                *wrappedObjectGVR,
+		wdsClientset:                    wdsClientset,
+		wdsDynamicClient:                wdsDynamicClient,
+		wdsName:                         wdsName,
+	}
+
+	transportController.logger.Info("Setting up event handlers")
+	// Set up an event handler for when PlacementDecision resources change
+	placementDecisionInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    transportController.enqueuePlacementDecision,
+		UpdateFunc: func(_, new interface{}) { transportController.enqueuePlacementDecision(new) },
+		DeleteFunc: transportController.enqueuePlacementDecision,
+	})
+	// Set up event handlers for when WrappedObject resources change. The handlers will lookup the origin PlacementDecision
+	// of the given WrappedObject and enqueue that PlacementDecision object for processing.
+	// This way, we don't need to implement custom logic for handling WrappedObject resources. More info on this pattern:
+	// https://github.com/kubernetes/community/blob/8cafef897a22026d42f5e5bb3f104febe7e29830/contributors/devel/controllers.md
+	wrappedObjectGenericInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    transportController.handleWrappedObject,
+		UpdateFunc: func(_, new interface{}) { transportController.handleWrappedObject(new) },
+		DeleteFunc: transportController.handleWrappedObject,
+	})
+	dynamicInformer.Start(ctx.Done())
+
+	return transportController, nil
+}
+
+func convertObjectToUnstructured(object runtime.Object) (*unstructured.Unstructured, error) {
+	unstructuredObject, err := runtime.DefaultUnstructuredConverter.ToUnstructured(object)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert given object to unstructured - %w", err)
+	}
+	return &unstructured.Unstructured{Object: unstructuredObject}, nil
+}
+
+func getGvrFromWrappedObject(restConfig *rest.Config, wrappedObject runtime.Object) (*schema.GroupVersionResource, error) {
```

Comment:
Returning a pointer here requires maintainers to think harder than return just the `schema.GroupVersionResource`. With return of a pointer _into_ a data structure returned from an upstream library, maintainers have to be sure that nothing assigns into that pointer or that such an assignment will not cause grief to the upstream library. In the error cases, this func could return a `schema.GroupVersionResource{}`.

Reply:
done